kong/router/atc_compat.lua

```diff
@@ -547,13 +564,62 @@ function _M:exec(ctx)
 
   req_uri = normalize(req_uri, true)
 
-  local match_t = self:select(req_method, req_uri, req_host, req_scheme,
-                              nil, nil, nil, nil,
-                              sni, headers)
+  local headers_key do
+    local headers_count = 0
+    for name, value in pairs(headers) do
+      local name = name:gsub("-", "_"):lower()
+
+      if type(value) == "table" then
+        for i = 1, #value do
+          value[i] = value[i]:lower()
+        end
+        tb_sort(value)
+        value = tb_concat(value, ", ")
```

Comment:
For hash key usage, I think it is unnecessary to concat with `, `, maybe a simple empty string?

Reply:
This was from the old router, it makes debugging a bit easier.